syntax = "proto3";

message Location {
    int32 person_id = 1;
    string longitude = 2;
    string latitude = 3;
    string creation_time = 4;
    int32 id = 5;
}

message Person {
    string company_name = 1;
    string last_name = 2;
    string first_name = 3;
    int32 id = 4;
}

message ConnectionMessage {
    Location location = 1;
    Person person = 2;
}

message ConnectionMessageList {
    repeated ConnectionMessage connections= 1;
}

message Contact {
    int32 person_id = 1;
    string start_date = 2;
    string end_date = 3;
    int32 meters = 4;
}

message Empty {

}

service ConnectionService {
    rpc Get(Empty) returns (ConnectionMessageList);
    rpc GetList(Contact) returns (ConnectionMessageList);
}